File: Juan/P2/Java/Napakalaki/src/napakalaki/Monster.java
package napakalaki;

/**
 *
 * @author juane
 * @version 2.1
 * Clase Monster
 */
public class Monster {
      
    /* Atributos propios de la clase*/
    
    private String name; // nombre del monstruo
    private int combatLevel; // nivel de combate del monstruo

    /* Relaciones atributo de tipo Prize y BadConsequence */
    private Prize prize; 
    private BadConsequence badConsequence;
     
    /* Constructor*/
    
    Monster(String n, int l, BadConsequence b, Prize p){
        
        this.name = n;
        this.combatLevel = l;
        this.badConsequence = b;
        this.prize = p;
    }
    
    /**
     * Método getName()
     * Devuelve el nombre del monstruo
     * @return name de tipo int
     */
    public String getName() {
        return name;
    }

    /**
     * Método getCombatLevel()
     * Devuelve el nivel de combate del monstruo
     * @return 
     */
    public int getCombatLevel() {
        return combatLevel;
    }

    /**
     * Método getBadConsequence()
     * Devuelve el mal royo
     * @return badConsequence de tipo BadConsequence
     */
    public BadConsequence getBadConsequence(){
        return this.badConsequence; 
    }
    
    /**
     * Método getLevelsGained()
     * Devuelve el número de niveles ganados proporcionados por su buen rollo
     * @return 
     */
    public int getLevelsGained(){         
        
        return this.prize.getLevels(); 
    }
    
    /**
     * Método getTreasuresGained()
     * Devuelve el número de tesoros ganados proporcionados por su buen rollo
     * @return 
     */
    public int getTreasuresGained(){        
        
        return this.prize.getTreasures(); 
    }
    
    
//    
//    /**
//     * Método getPrice()
//     * Devuelve el buen royo
//     * @return  price de tipo Prize
//     */
//    public Prize getPrice() {
//        return price;
//    }

    
    /**
     * Método toString
     * @return 
     */
    @Override
    public String toString() {
        return "Monster = " + this.name + ", combatLevel = " 
                +  this.combatLevel + ", Price: " +  this.prize 
                + ", " + this.badConsequence ;
    }
}

File: Juan/P3/Ruby/Napakalaki/lib/Monster.rb
#encoding: utf-8

#Versión 3.0

require_relative 'TreasureKind.rb'
require_relative 'BadConsequence.rb'
require_relative 'Prize.rb'



module NapakalakiGame
class Monster
    # @@numMonster --> contar el número de monstruos
    #Atributos propios de la clase
  
    # @name # String para representar el nombre del monstruo
    # @combatLevel # int para representar el nivel de combate del monstruo
  
    #Otros atributos
    # @badconsequence  # de tipo BadConsequence
    # @price # de tipo Prize
 
    
  
  
    def initialize(name,level,bc,price)
        @name = name
        @combatLevel = level    
        @prize = price
        @badconsequence = bc
    
    end
  
    ## Métodos get
    attr_reader :name
    attr_reader :combatLevel 
    attr_reader :badconsequence
    attr_reader :prize
    
   
    
    def getLevelsGained
        @prize.levels
    end
    
    def getTreasureGained
        @prize.treasures
    end

    #EXAMEN
    
    def getCombatLevel
        dice = Dice.instance  
        cardealer = CardDealer.instance
        
        friend = self
        
       
        r = dice.NextNumber
        if r > 5
            cardealer.getRandomUnusedMonster
            
        end
        
        if friend != nil
          cl =  friend.getCombatLevel
             
        else
            cl = 0
         return cl  # monsterLelevl
        end
         
    end
    #FIN EXAMEN
    def to_s
        "Monster =  #{@name}  , combatLevel =   #{@combatLevel} ,  #{@badconsequence} ,  Price:  #{@prize} "      
    end
  
end
end